### PR TITLE
Update PR template for KDS release process

### PR DESCRIPTION
**Pull request by sairina**
## Description

Updates the PR template with a section for the PR-writer to add in the `changelog` and also to manually point the most recent version of KDS to the Kolibri `develop` branch.

#### Issue addressed

Addresses #247

### Does this introduce any tech-debt items?

We will need to make sure that there is clear documentation about how to add to the `changelog` (this may be something we can add to the PR template in the future so that anyone can just access it). For now, we can just tell folks how to do it (internally). Same goes with how to point Kolibri to the right branch of KDS. 

## Reviewer guidance
- [ ] Is the code clean and well-commented?

## Comments

Doesn't need to be said, but all feedback appreciated, of course.

### CODE REVIEW

File: .github/pull_request_template.md
Comment (indirectlylit) [inline .github/pull_request_template.md:36]: Should we require similar PRs for Studio and KDP?

If that's too onerous, maybe we should actually be looking into auto-generated PRs based on github actions?

Comment (sairina) [reply .github/pull_request_template.md:36]: Hmm. Good point. Should we ask the folks in charge of those projects (or at least the release managers)?

Comment (indirectlylit) [reply .github/pull_request_template.md:36]: posted in Slack here: https://learningequality.slack.com/archives/C01PBEMPZC7/p1625172143079600

Comment (sairina) [reply .github/pull_request_template.md:36]: That makes a lot of sense! I'll update it to reflect that.

Comment (indirectlylit) [reply .github/pull_request_template.md]: I'm a little confused on one detail:

If some KDS PR has not been merged yet, what commit would the product update PRs point at? I was expecting that the update PRs would point to the merge commit of the KDS PR.

Comment (MisRob) [reply .github/pull_request_template.md]: That makes sense but then the question is if it should be in "Reviewer guidance" of the PR template when it's not possible to check it during the review process itself

Comment (MisRob) [reply .github/pull_request_template.md]: (I wrote my first note assuming that it's something to be done during the review)

Comment (indirectlylit) [reply .github/pull_request_template.md]: Ah, that's a good point.

Maybe we should add a post-merge checklist? e.g. something like...

> # Post-merge updates
> 
> After merging, unstabled branches of Kolibri products (Learning Platform, Studio, and Data Portal) should be updated to point at the merge commit resulting from this PR. This process should be led by the submitter of the Design System PR in collaboration with other LE team members working on the other product repos.
>  
> * [ ] Learning Platform update PR is submitted
> * [ ] Studio update PR is submitted
> * [ ] Data Portal update PR is submitted
> * [ ] Learning Platform update PR is merged
> * [ ] Studio update PR is merged
> * [ ] Data Portal update PR is merged


thoughts?

Comment (MisRob) [reply .github/pull_request_template.md]: Makes much sense to me

Comment (MisRob) [reply .github/pull_request_template.md]: I appreciate that each product has its own item on the list, could be helpful to keep track of what still needs to be done.

Comment (indirectlylit) [reply .github/pull_request_template.md]: FYI, after discussing in Product Tactical, @rtibbles decided that we should not consider the "merged" items mandatory, at least in the shorter-term. We can still include the full list, but the most important things are

1. keeping the changelog up-to-date
2. opening PRs against target products for visibility

Newer updates should supersede older ones in the case where multiple KDS have happened before a product is updated with the latest.

(Longer-term, we can automate opening of PRs with github actions, similar to dependabot)

Comment (indirectlylit) [reply .github/pull_request_template.md]: We may want to update the template text to clarify the above

File: .github/pull_request_template.md
Comment (indirectlylit) [inline .github/pull_request_template.md]: Is this item intended to be checked before or after the post-merge updates? It seems to describe similar tasks as

```
- [ ] Learning Platform update PR is submitted
- [ ] Learning Platform update PR is merged
```

Comment (sairina) [reply .github/pull_request_template.md]: Ah yes. Let's get rid of this.

**Review by indirectlylit — APPROVED** [review]
awesome, thank you!

**Pull request merged**
